thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js

```diff
@@ -341,117 +145,141 @@ export default Component.extend({
    * }
    */
   links: computed(
-    'urns',
+    'entities',
     function() {
-      const { urns, entities } = this.getProperties('urns', 'entities');
+      const { entities } = this.getProperties('entities');
       let metricUrlMapping = {};
 
-      urns.forEach(urn => {
-        const attributes = entities[urn].attributes;
-        const { externalUrls = [] } = attributes;
-        let urlArr = [];
-
-        // Add the list of urls for each url type
-        externalUrls.forEach(urlLabel => {
-          urlArr.push({
-            [urlLabel]: attributes[urlLabel][0] // each type should only have 1 url
+      filterPrefix(Object.keys(entities), 'thirdeye:metric:')
+        .forEach(urn => {
+          const attributes = entities[urn].attributes;
+          const { externalUrls = [] } = attributes;
+          let urlArr = [];
+
+          // Add the list of urls for each url type
+          externalUrls.forEach(urlLabel => {
```

Comment:
looks like `reduce` might be a good candidate here since we have transforming n to 1

Reply:
yes. code moved only